src/login/logind-dbus.c

```diff
@@ -1835,11 +1835,14 @@ static int method_do_shutdown_or_sleep(
 
         if (sleep_verb) {
                 r = can_sleep(sleep_verb);
+                if (r == -ENOSPC)
+                        return sd_bus_error_set(error, BUS_ERROR_SLEEP_VERB_NOT_SUPPORTED,
```

Comment:
i wonder if this should get a new BUS_ERROR_xyz code?

Reply:
Dunno. If you think it should, I can add it, but I didn't see a reason atm.